alibi_detect/utils/loading.py

```diff
@@ -0,0 +1,625 @@
+# TODO - Need to modularise torch and tensorflow imports and use. e.g. has_tensorflow and has_pytorch etc
+from alibi_detect.utils.registry import registry
+from alibi_detect.utils.schemas import SupportedModels  # type: ignore[attr-defined]
+from alibi_detect.utils.validate import validate_config
+from alibi_detect.utils.tensorflow._loading import load_model as load_model_tf, \
+    prep_model_and_emb as prep_model_and_emb_tf, load_kernel_config as load_kernel_config_tf, \
+    load_optimizer as load_optimizer_tf, load_embedding as load_embedding_tf, load_detector_legacy, Detectors
+#  from alibi_detect.utils.pytorch._loading import set_device
+import tensorflow as tf  # TODO - this is currently only required for FIELDS_TO_DTYPE conversion. Remove in future.
```

Comment:
The `tensorflow` import will need to be removed in future in order to allow tensorflow to be an optional depedency.